flag_engine/identities/models.py

```diff
@@ -51,6 +63,35 @@ def in_segment(self, segment: SegmentModel) -> bool:
             for rule in segment.rules
         )
 
+    def find_trait_index(self, trait: TraitModel) -> typing.Optional[int]:
+        try:
+            return self.identity_traits.index(trait)
+        except ValueError:
+            return None
+
+    def update_traits(self, traits: typing.List[TraitModel]):
```

Comment:
I think we could improve this method a little if we convert the traits up front into a dictionary of `trait.trait_key: trait`? It would make things easier and we wouldn't need the `find_trait_index` function which I think is a bit nasty. 

Alternatively, could we use a set instead?

Reply:
>Alternatively, could we use a set instead?
convert `identity_traits` to a set instead of a list? I don't that is going to be of much use anyway, we still need to find the index of the trait to remove and update.